source/extensions/filters/http/dynamic_forward_proxy/proxy_filter.h

```diff
@@ -20,11 +20,13 @@ class ProxyFilterConfig {
 
   Extensions::Common::DynamicForwardProxy::DnsCache& cache() { return *dns_cache_; }
   Upstream::ClusterManager& clusterManager() { return cluster_manager_; }
+  bool saveUpstreamAddress() const { return save_upstream_address_; };
 
 private:
   const Extensions::Common::DynamicForwardProxy::DnsCacheManagerSharedPtr dns_cache_manager_;
   const Extensions::Common::DynamicForwardProxy::DnsCacheSharedPtr dns_cache_;
   Upstream::ClusterManager& cluster_manager_;
+  bool save_upstream_address_ = false;
```

Comment:
Make this const and remove the initializer as you you always initialize it in the constructor.